enos/provider/g5k.py

```diff
@@ -218,56 +273,88 @@ def _check_nodes(self,
             wanted_nodes += reduce(operator.add, map(int, roles.values()))
 
         if mode == ROLE_DISTRIBUTION_MODE_STRICT and wanted_nodes > len(nodes):
-            raise Exception("Not enough nodes to continue")
+            raise Exception("Not enough nodes to continue (%d missing)"
+                    % (wanted_nodes - len(nodes)))
 
         return True
 
-    def _get_job(self):
-        """Get the hosts from an existing job (if any) or from a new job.
-        This will perform a reservation if necessary."""
+    def _get_job_ids(self):
```

Comment:
The same as `_get_queues`, I would pass the `config` and try to avoid any side effects (`self.gridjobs`). This is a design decision that has to be discussed and that goes beyond the current PR. More generally, many methods in the provider are only utils functions that could be extracted elsewhere. 
(cc @rcherrueau)